crypto/evp/kdf_lib.c

```diff
@@ -144,6 +144,36 @@ int EVP_KDF_derive(EVP_KDF_CTX *ctx, unsigned char *key, size_t keylen,
     return ctx->meth->derive(ctx->algctx, key, keylen, params);
 }
 
+EVP_SKEY *EVP_KDF_derive_SKEY(EVP_KDF_CTX *ctx, EVP_SKEYMGMT *skeymgmt,
```

Comment:
The EVP_SKEYMGMT *skeymgmt argument should probably be replaced by a "key type" const char * argument with the expect key type the derive function should produce (eg. "AES").

Alternatively the argument can be move to a mandatory param.

That information needs to be passed into meth->derive_opaque() again either as an explicit argument or as a param.

Reply:
I need a EVP_SKEYMGMT to free allocated blob.
I can either delegate creation to application (and pass already allocated keymgmt object) or to provider (and implement some API for this purpose, then I will need some argument, such as `const char * key_type`)

Both options have their downsides.